inc/saitam.h

```diff
@@ -0,0 +1,883 @@
+ * @param[in] attr attribute to set
+ *
+ * @return #SAI_STATUS_SUCCESS on success Failure status code on error
+ */
+typedef sai_status_t(*sai_tam_snapshot_attribute_set_fn)(
+        _In_ sai_object_id_t snapshot_id,
+        _In_ const sai_attribute_t *attr);
+
+/**
+ * @brief Get values for specified Snapshot attributes.
+ *
+ * @param[in] snapshot_id Snapshot object id
+ * @param[in] attr_count number of attributes
+ * @param[inout] attr_list array of attributes
+ *
+ * @return #SAI_STATUS_SUCCESS on success Failure status code on error
+ */
+typedef sai_status_t(*sai_tam_snapshot_attribute_get_fn)(
+        _In_ sai_object_id_t snapshot_id,
+        _In_ uint32_t attr_count,
+        _Inout_ sai_attribute_t *attr_list);
+
+/**
+ * @brief Obtain the values for all statistics from a snapshot.
+ *
+ * Attribute list must supply sufficient memory for statistics
+ * as specified for the snapshot object, which may be all statistics
+ * supported by the associated tam object.
+ *
+ * @param[in] snapshot_id Snapshot object id
+ * @param[inout] stat_count Number of statistics (required/provided)
+ * @param[inout] statistics Statistics (allocated/provided)
+ *
+ * @return #SAI_STATUS_SUCCESS on success Failure status code on error
+ */
+typedef sai_status_t(*sai_tam_snapshot_statistics_get_fn)(
+        _In_ sai_object_id_t snapshot_id,
+        _Inout_ uint32_t *stat_count,
+        _Inout_ sai_tam_statistic_t *statistics);
+
+/**
+ * @brief TAM type of snapshot transport
+ */
+typedef enum _sai_tam_transporter_type_t
+{
+
+    /** Transport local, to the CPU */
+    SAI_TAM_TRANSPORTER_TYPE_LOCAL,
+
+    /** Transport remote, to a remote monitoring client */
+    SAI_TAM_TRANSPORTER_TYPE_REMOTE,
+
+} sai_tam_transporter_type_t;
+
+/**
+ * @brief TAM Snapshot Transporter Attributes
+ */
+typedef enum _sai_tam_transporter_attr_t {
+    /**
+     * @brief Start of Attributes
+     */
+    SAI_TAM_TRANSPORTER_ATTR_START,
+
+    /**
+     * @brief Transporter Type
+     *
+     * If this attribute value is unspecified the local transporter is used.
+     *
+     * @type sai_tam_transporter_type_t
+     * @flags CREATE_AND_SET
+     * @default SAI_TAM_TRANSPORTER_TYPE_LOCAL
+     */
+
+    SAI_TAM_TRANSPORTER_ATTR_TYPE = SAI_TAM_TRANSPORTER_ATTR_START,
+
+    /**
+     * @brief Maximum size beyond which it will be truncated.
+     *
+     * If this attribute value is zero or unspecified, snapshots are not
+     * truncated while transporting. (DEFAULT)
+     *
+     * @type sai_uint32_t
+     * @flags CREATE_AND_SET
+     * @default 0
+     */
+    SAI_TAM_TRANSPORTER_ATTR_MAX_SNAPSHOT_SIZE,
+
+    /**
+     * @brief Mirroring session object defining the remote transport capabilities.
+     *
+     * If this attribute is unspecified, Local CPU Transport is used (DEFAULT)
+     *
+     * @type sai_object_id_t
+     * @flags CREATE_AND_SET
+     * @objects SAI_OBJECT_TYPE_TAM_TRANSPORTER
+     * @allownull true
```

Comment:
I tried, but am getting a lot of other error with the metadata code itself. Will debug that later.